librz/core/cmd.c

```diff
@@ -6008,17 +5830,22 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(html_enable_stmt) {
 }
 
 DEFINE_HANDLE_TS_FCN_AND_SYMBOL(pipe_stmt) {
-	TSNode first_cmd = ts_node_named_child(node, 0);
-	rz_return_val_if_fail(!ts_node_is_null(first_cmd), false);
-	TSNode second_cmd = ts_node_named_child(node, 1);
-	rz_return_val_if_fail(!ts_node_is_null(second_cmd), false);
-	char *first_str = ts_node_sub_string(first_cmd, state->input);
-	char *second_str = ts_node_sub_string(second_cmd, state->input);
-	int value = state->core->num->value;
-	RzCmdStatus res = rz_cmd_int2status(rz_core_cmd_pipe(state->core, first_str, second_str));
-	state->core->num->value = value;
-	free(first_str);
-	free(second_str);
+	TSNode command_rizin = ts_node_named_child(node, 0);
+	TSNode command_pipe = ts_node_named_child(node, 1);
+
+	char *rz_cmd = ts_node_sub_string(command_rizin, state->input);
```

Comment:
This is probably unnecessary. We can probably use `handle_ts_stmt` as done elsewhere to direclty work with the TSNode, so we avoid going again through the whole tree-sitter parsing and handling.